File: Day219_2025/Q219_GFG.java
import java.util.*;

class Q219_GFG {
    public int minDifference(String[] arr) {
        List<Integer> op = new ArrayList<>();
        
        for (String time : arr) {
            int hr = 3600 * Integer.parseInt(time.substring(0, 2));
            int mint = 60 * Integer.parseInt(time.substring(3, 5));
            int sec = Integer.parseInt(time.substring(6, 8));
            
            int total = hr + mint + sec;
            total = total % 86400;  // seconds in a day
            op.add(total);
        }

        Collections.sort(op);
        int n = op.size();
        int ans = Integer.MAX_VALUE;

        for (int i = 1; i < n; i++) {
            ans = Math.min(ans, op.get(i) - op.get(i - 1));
        }

        ans = Math.min(ans, (86400 + op.get(0) - op.get(n - 1)) % 86400);
        return ans;
    }
}

File: Day219_2025/Q219_GFG.py
class Solution:
    def minDifference(self, arr):
        op = []

        for time in arr:
            hr = 3600 * int(time[0:2])
            mint = 60 * int(time[3:5])
            sec = int(time[6:8])
            
            total = (hr + mint + sec) % 86400
            op.append(total)

        op.sort()
        n = len(op)
        ans = float('inf')

        for i in range(1, n):
            ans = min(ans, op[i] - op[i - 1])
        
        ans = min(ans, (86400 + op[0] - op[n - 1]) % 86400)
        return ans
